### PR TITLE
Feat/backup rather than hash check

### PR DESCRIPTION
**Pull request by andywaltlova**
[HMS-3039](https://issues.redhat.com/browse/HMS-3039) , [HMS-3038](https://issues.redhat.com/browse/HMS-3038)

Text of issue:

> If system has older /etc/yum.repos.d/convert2rhel.repo or  /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release. Then instead of providing message "Hash mismatch between the downloaded file and the one present on the system"
>
> Can we change wording of message somewhat like "Older convert2rhel files exist on the system, remove them and rerun analysis"
> 
> Furthermore, we can also specify to specifically check below two file and remove them
> 
> /etc/yum.repos.d/convert2rhel.repo
> 
> /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release

Talked with @bocekm in the comments, and we could change our approach to backup the files instead and restore them after script completes, that way we will get rid of one error and spare users of the troubles for remediating it

- [ ] Make exception for GPG key? and In case of successful conversion keep it for new rhel system?

### CODE REVIEW

File: scripts/conversion_script.py [hunk 1]
```diff
@@ -292,32 +274,39 @@ def cleanup(required_files):
...
+            )
+            os.rename(required_file.path, required_file.path + ".backup")
+    except (IOError, OSError) as exception:
+        # pylint: disable=raise-missing-from
+        # This is expected and we want to control the flow with ProcessError
+        # instead of re-raising the exception.
+        raise ProcessError(
+            message="An unexpected error occured during restore or backup of required file.",
+            report=str(exception),
+        )
```
Comment (r0x0d) [inline scripts/conversion_script.py:309]: Might be better to let the exception fall through and be caught in the main exception.

Comment (andywaltlova) [reply scripts/conversion_script.py:309]: ah right, that would be easier .. probably though about it too much :sweat_smile:

**Review by r0x0d — APPROVED** [review]
looks awesome from my point of view.

**Comment by r0x0d** [comment]
> Make exception for GPG key? and In case of successful conversion keep it for new rhel system?

I would say: yes! 

In case we have a successful conversion, let's leave the gpg key.

**Comment by andywaltlova** [comment]
> > Make exception for GPG key? and In case of successful conversion keep it for new rhel system?
> 
> I would say: yes!
> 
> In case we have a successful conversion, let's leave the gpg key.

Added that feature in latest commit https://github.com/oamg/convert2rhel-worker-scripts/pull/30/commits/2a1a59c38c432aa743ea7a207356e57df5e10473 , check it out

**Pull request merged**